### PR TITLE
Fix auto-complete in slash command for shared channel invite

### PR DESCRIPTION
**Pull request by wiggin77**
#### Summary
A recent [PR](https://github.com/mattermost/mattermost/pull/24877) added a context var to many slash command methods.  One was missed that broke the auto-complete of slash command to set remoteID when sharing a channel.  The break was due to the shared channels auto-complete provider no longer implementing the correct interface.

This PR fixes the broken auto-complete and adds a compile time type check to ensure it doesn’t break again.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-55718

#### Release Note
```release-note
NONE
```

### CODE REVIEW

File: server/channels/app/slashcommands/command_share.go [hunk 1]
```diff
@@ -17,6 +17,9 @@ import (
 type ShareProvider struct {
 }
 
+// ensure ShareProvide implements AutocompleteDynamicArgProvider
+var _ app.AutocompleteDynamicArgProvider = (*ShareProvider)(nil)
```
Comment (hanzei) [inline server/channels/app/slashcommands/command_share.go:21]: :100:

**Review by hanzei — APPROVED** [review]
Good catch, thanks for fixing my mistake. :+1: 

Should this PR go into v9.3?

**Comment by wiggin77** [comment]
> Good catch, thanks for fixing my mistake. :+1: 
> 
> Should this PR go into v9.3?

Not your mistake - there was no way to know that provider was broken. I should have put a type-check in there originally.

I think next release is fine. Shared Channels is not GA yet.

**Review by mgdelacroix — COMMENTED** [review]
LGTM! :+1:

**Review by mgdelacroix — APPROVED** [review]

**Pull request merged**